etna/models/sarimax.py

```diff
@@ -256,30 +268,36 @@ def predict(self, df: pd.DataFrame, prediction_interval: bool, quantiles: Sequen
             )
 
         exog_future = self._select_regressors(df)
+        start_timestamp = df["timestamp"].min()
+        end_timestamp = df["timestamp"].max()
+        start_idx = determine_num_steps(
+            start_timestamp=self._first_train_timestamp, end_timestamp=start_timestamp, freq=self._freq  # type: ignore
+        )
+        end_idx = determine_num_steps(
+            start_timestamp=self._first_train_timestamp, end_timestamp=end_timestamp, freq=self._freq  # type: ignore
+        )
```

Comment:
May we should add assert there, something like
`end_idx-start_idx == len(df), "Check that total number of steps to forecast is equal to total ts lenght` and so so on

It's not obvious how number of steps  has become index